lib/main.dart

```diff
@@ -144,11 +145,16 @@ class PerfToolFramework extends Framework {
       try {
         reloadAction.disabled = true;
         await serviceManager.performHotReload();
+        messageBus.addEvent(BusEvent('reload.start'));
         timer.stop();
         // 'reloaded in 600ms'
-        status.setText('reloaded in ${nf.format(timer.elapsedMilliseconds)}ms');
+        final String message = 'reloaded in ${_renderDuration(timer.elapsed)}';
```

Comment:
With this change reloads/restarts triggered by the flutter tools console app or an IDE will now act differently than ones triggered by the devtools app.

If this wiring was instead done on the flutter tools side when a reload or restart event occurred we could have consistent behavior regardless of where the event was triggered.

Reply:
> With this change reloads/restarts triggered by the flutter tools console app or an IDE will now act differently than ones triggered by the devtools app.

Good point

> If this wiring was instead done on the flutter tools side when a reload or restart event occurred we could have consistent behavior regardless of where the event was triggered.

I'll do some investigation into how we could notify from flutter_tools (and if so how this PR would change). I'd definitely be better to have flutter_tools be the system of record (and, to get stderr from it as well).

Reply:
To follow up, some options are:

- to add a reflector API in package:flutter: `ext.flutter.multicastEvent`
- to add a reflector API in dart:developer `fireMulticastEvent`
- add a simple pub-sub set of APIs in dart:developer (though, `fireMulticastEvent` may just be that?)

Reply:
all these options seem good. I'm happy to land this as is with a TODO to switch to a multicast solution in the future. This code is still in the right direction and will make it easier to switch to multiCast once flutter or dart:developer supports it.

Reply:
added a todo; will land, and move towards having flutter_tools send hot reload events in some manner in the future